panoramapublic/src/org/labkey/panoramapublic/PanoramaPublicController.java

```diff
@@ -725,6 +738,138 @@ public Container getSupportContainer()
         }
     }
 
+    /*
+    Set the user account that can be used to download public datasets on Panorama Public with a WebDAV client such as
+    Cyberduck, or by mapping a network drive in Windows. Anonymous downloads do not work with WebDAV.
+     */
+    @RequiresPermission(AdminOperationsPermission.class)
+    public static class AddPublicDataUserAction extends FormViewAction<PublicDataUseForm>
+    {
+        private Journal _journal;
+        @Override
+        public void validateCommand(PublicDataUseForm form, Errors errors)
+        {
+            _journal = form.lookupJournal();
+            if (_journal == null)
+            {
+                errors.reject(ERROR_MSG, "No journal found for journal Id " + form.getId());
+            }
+            if (StringUtils.isBlank(form.getUserEmail()))
+            {
+                errors.reject(ERROR_MSG, "User email cannot be blank");
+            }
+            if (StringUtils.isBlank(form.getUserPassword()))
+            {
+                errors.reject(ERROR_MSG, "User password cannot be blank");
+            }
+        }
+
+        @Override
+        public ModelAndView getView(PublicDataUseForm form, boolean reshow, BindException errors)
+        {
+            if (!reshow)
+            {
+                _journal = form.lookupJournal();
+                if (_journal == null)
+                {
+                    errors.reject(ERROR_MSG, "No journal found for journal Id " + form.getId());
+                    return new SimpleErrorView(errors);
+                }
+                JournalManager.PublicDataUser publicDataUser = JournalManager.getPublicDataUser(_journal);
+                if (publicDataUser != null)
+                {
+                    form.setUserEmail(publicDataUser.getEmail());
+                }
+            }
+
+            HtmlView view = new HtmlView(
+                    DIV(
+                            ERRORS(errors),
+                            "Enter the email address for the user account that can be used to download public datasets in a WebDAV client",
+                            FORM(at(method, "POST", action, new ActionURL(AddPublicDataUserAction.class, getContainer()).addParameter("id", _journal.getId())),
```

Comment:
Why is it important to let the admin choose their own username/password here? Couldn't everyone use the public@proteinms.net account?

Reply:
This is a one-time setup for a site admin to choose the user account that can be used for downloading files in a WebDAV client.  The user Id and password are saved in an encrypted store, and are used when rendering dataDownloadInfo.jsp.

Reply:
Got it. Makes sense. I was thinking this was being configured at the per-folder level.